File: 2537. Count the Number of Good Subarrays/Solution.py
from collections import defaultdict


class Solution:
    def countGood(self, nums: list[int], k: int) -> int:
        cnt = defaultdict(int)
        ans = good_cnt = left = 0
        for x in nums:
            good_cnt += cnt[x]
            cnt[x] += 1
            while good_cnt >= k:
                cnt[nums[left]] -= 1
                good_cnt -= cnt[nums[left]]
                left += 1
            ans += left
        return ans

File: 2537. Count the Number of Good Subarrays/Solution.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    long long countGood(vector<int>& nums, int k)
    {
        unordered_map<int, int> cnt;
        int i = 0, n = nums.size();
        long long good_cnt = 0, ans = 0;
        for (int j = 0; j < n; ++j) {
            good_cnt += cnt[nums[j]]++;
            while (good_cnt >= k) {
                good_cnt -= --cnt[nums[i++]];
                ans += n - j;
            }
        }
        return ans;
    }
};
